Rename heat conduction to heat transfer

GiudGiud opened the pull request:
let s see how the apps do
the one tedious thing could be if they link to doco pages in the module

refs https://github.com/idaholab/moose/issues/25584

Recipe patches needed:
https://github.com/idaholab/civet_recipes/pull/1462
- Framework 1
- Modules
- Modules parallel
- Modules recover
- Install combined

App patches needed:
- [x] Marmot https://github.inl.gov/ncrc/marmot/pull/910
- [x] Bison https://github.inl.gov/ncrc/bison/pull/5619
- [x] Grizzly https://github.inl.gov/ncrc/grizzly/pull/742
- [x] Pronghorn https://github.inl.gov/ncrc/pronghorn/pull/1224
- [x] Blue Crab https://github.inl.gov/ncrc/blue_crab/pull/224
- [x] Direwolf https://github.inl.gov/ncrc/dire_wolf/pull/282
- [x] Griffin https://github.inl.gov/ncrc/griffin/pull/1526
- [x] Relap7 https://github.inl.gov/ncrc/relap-7/pull/2548
- [x] SAM https://git-nse.egs.anl.gov/SAM/SAM/-/merge_requests/828
- [x] Sockeye https://github.inl.gov/ncrc/sockeye/pull/728
- [x] BlackBear https://github.com/idaholab/blackbear/pull/361
- [x] Isopod https://github.com/idaholab/isopod/pull/83
- [x] Magpie https://github.com/idaholab/magpie/pull/490
- [x] Malamute https://github.com/idaholab/malamute/pull/104
- [x] Mastodon https://github.com/idaholab/mastodon/pull/437
- [x] TMAP8 https://github.com/idaholab/TMAP8/pull/72

GiudGiud committed:
Rename heat conduction module to heat transfer, refs #25584

GiudGiud committed:
Adapt modules makefile for new name and add a warning

GiudGiud committed:
Update SQA and config reports to new module name

GiudGiud committed:
Update paths in doco to new module (and executable) name

GiudGiud committed:
Change module name for executable

GiudGiud committed:
Adapt to new module name in tests and tutorials

GiudGiud committed:
Adapt module loader

GiudGiud committed:
Adapt app creation code to new name

GiudGiud committed:
Adapt yml config docs to new name

GiudGiud committed:
Adapt moosedocs python to new module name

GiudGiud committed:
Adapt SQA doc and some tester doc pages to new module name

GiudGiud committed:
Adapt stork and tutorials to new name

GiudGiud committed:
adapt gitignore to new name

GiudGiud committed:
Re-name documentation files for renamed module

GiudGiud committed:
Adapt path to heat conduction items in MOOSE doco
Adapt list of modules

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1791557/) on 353d788 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25585/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25585/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format fe7907a516e64373d9285656d223a34ea58668f8`

GiudGiud committed:
Address testing failures

GiudGiud committed:
Adapt registration of source objects

GiudGiud committed:
Adapt test app

GiudGiud committed:
Change up syntax doco for HeatConduction action under Modules

GiudGiud committed:
Adapt other module sqa docs to new module name

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1792542/) on 30d274b wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/25585/site)

This comment will be updated on new commits.

GiudGiud commented:
I think we should get this in like:
- get the green light from a reviewer here
- merge the civet recipe change (which breaks next already)
- make sure it s green on those
- merge this PR (unbreaks previous break, breaks due to apps)
- merge all the app patches

lindsayad commented on modules/heat_transfer/doc/content/modules/heat_transfer/sqa/heat_transfer_far.md:
```diff
@@ -0,0 +1 @@
+!template load file=sqa/module_far.md.template category=heat_transfer module=Heat Conduction
```
Maybe you are missing "Heat Conduction" -> "Heat Transfer"?

GiudGiud replied:
good catch

GiudGiud replied:
fixed

GiudGiud committed:
Heat Conduction -> Heat Transfer, only where it makes sense

loganharbour commented:
This has a bit to go. Still need to do the makefile changes, and should probably be able to get away with not immediately breaking everyone by setting up a symlink in modules

GiudGiud commented:
All the apps build properly. We only break their documentation because the sqa docs are all hardcoded. 

I changed a lot of makefiles already. I don’t know which one is missing changes

GiudGiud committed:
Rename heat conduction module to heat transfer module is misc doco pages

GiudGiud committed:
Add symlink to the heat conduction modules
This broke tests last time, but maybe we can work around it and get apps to pass their doco recipe

GiudGiud commented:
Making symlinks to the heat_conduction module got 3 more recipes green.
I think we can proceed with the 5-step plan now

lindsayad commented on modules/heat_conduction/include:
Shouldn't we just need the top-level symlink?

GiudGiud replied:
If you do that you run the tests twice in the same folders and that crashes the 3 recipes I mentione

lindsayad commented on modules/heat_transfer/src/base/HeatConductionApp.C:
```diff
@@ -0,0 +1,10 @@
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "HeatTransferApp.h"
```
Seems like this file is unnecessary. But if it's necessary
```suggestion
#include "HeatConductionApp.h"
```

GiudGiud replied:
yeah that s what I was trying to do
I want that header to be included at least once so it s compiled

lindsayad commented on modules/heat_transfer/test/include/base/HeatConductionTestApp.h:
```diff
@@ -0,0 +1,17 @@
+#include "MooseApp.h"
+#include "HeatTransferTestApp.h"
+
+class HeatTransferTestApp : public HeatConductionTestApp
```
```suggestion
class HeatConductionTestApp : public HeatTransferTestApp
```

lindsayad commented on modules/module_loader/include/ModulesApp.h:
```diff
@@ -35,6 +35,9 @@
 #ifdef HEAT_CONDUCTION_ENABLED
 #include "HeatConductionApp.h"
 #endif
+#ifdef HEAT_TRANSFER_ENABLED
+#include "HeatTransferApp.h"
+#endif
```
So is this part of the 5 step plan having these both together?

GiudGiud replied:
the 3 lines above those will be removed

lindsayad commented on modules/module_loader/include/ModulesApp.h:
```diff
@@ -135,8 +138,8 @@ ModulesApp::registerAllObjects(Factory & f, ActionFactory & af, Syntax & s)
   GeochemistryApp::registerAll(f, af, s);
 #endif
 
-#ifdef HEAT_CONDUCTION_ENABLED
-  HeatConductionApp::registerAll(f, af, s);
```
Seems like if you are going to keep the above lines, then you should keep these as well

GiudGiud replied:
this code is performed by inheritance we dont need to redefine it

lindsayad commented on modules/module_loader/src/ModulesApp.C:
My same questions from the header apply here

lindsayad commented on modules/modules.mk:
```diff
@@ -62,19 +62,25 @@ endif
 endif
 
+# heat_conduction was renamed to heat_transfer
 ifeq ($(HEAT_CONDUCTION),yes)
```
Is it possible for this to be yes considering your line removals above?

GiudGiud replied:
if set externally

GiudGiud replied:
also app makefiles still refer to heat_conduction in their makefile

GiudGiud committed:
Apply suggestions from code review

Co-authored-by: Alex Lindsay <alexander.lindsay@inl.gov>

lindsayad reviewed (dismissed):
This looks ok to me, but it would be nice to have integration sign-off from @loganharbour

loganharbour reviewed (dismissed):
- Need to fix the installed dirs for combined, see https://civet.inl.gov/job/1795152/
- recipe PR isn't passing

GiudGiud committed:
Exclude heat conduction from installable dirs for combined

GiudGiud commented:
The installed dirs for combined was fooled by the symlink

looks like civet_recipe is hosed right now. My PR was passing fine

loganharbour commented:
yeah, the SSH issues to github are also on HPC. definitely best to wait for that to be resolved

loganharbour commented:
There's an env update that's been stuck on next, which definitely needs to get to moose master and then to all app masters. Next hasn't merged in a bit. I would definitely suggest waiting for all of that to propagate through before this train. At least fully through the evening, assuming the SSH issues get resolved. good otherwise

GiudGiud commented:
Sounds good. 

Rebased to handle conflict

CIVET recipes work again (green on testing)
Next has merged two hours ago https://civet.inl.gov/job/1796527/

loganharbour commented:
SSH access to github.com is still broken, merge events on civet are still very flaky. talk to @milljm about the ticket being completed

lindsayad reviewed (changes requested):
blocking while we get feedback from the community

GiudGiud committed:
Attempt fooling app SQA doc recipe with symlink

GiudGiud commented:
Just got an idea that might reduce app failures. Let s see.
It s essentially a double symlink

GiudGiud commented:
no luck on fooling the apps doco builds

GiudGiud commented:
Let's go?

lindsayad commented:
Based on slack it sounds like @loganharbour wants to look at the CI results

moosebuild commented:
Job [Controlled app tests](https://civet.inl.gov/job/1810889/) on 40c2dc6 : invalidated by @lindsayad

moosebuild commented:
Job [OpenMPI](https://civet.inl.gov/job/1810901/) on 40c2dc6 : invalidated by @lindsayad

GiudGiud commented:
I'll wait for github.inl.gov & co (seems there s a wednesday thing too) to be back in working order but thanks for the approval!

loganharbour commented:
Why are the names in the makefiles not changed in the app patches?

GiudGiud commented:
It still builds with the old name, we can have it either way during the transition phase.

GiudGiud committed:
Clearer docstring in Couplable second derivative

loganharbour commented:
> It still builds with the old name, we can have it either way during the transition phase.

why would we not do it all at once?

GiudGiud commented:
I can do it when I update the submodule

loganharbour commented:
See message in slack. Good to go at 9 MST
